src/kernels/jupyter/launcher/serverUriStorage.ts

```diff
@@ -56,6 +56,9 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage, IServe
     public get onDidChange(): Event<void> {
         return this._onDidChangeUri.event;
     }
+    public get onDidChangeConnectionType(): Event<void> {
```

Comment:
It's duplicate but intentional. We want to get rid of `IServerConnectionType` and one approach is merging it into `ServerUriStorage#onDidChangeConnectionType`. I didn't do a clean up here as it will make this PR touch dozen more files.